safe-global/safe-wallet-web
# feat: Add support for proposing transactions as a delegate

## What it solves

Resolves #1825

## How this PR fixes it

- Check if a connected wallet is a delegate and store it as a flag in `useWallet`
- Enable transaction buttons for delegates
- Display a `DelegateForm` when creating a transaction and only propose the transaction

## ToDos

- [x] The transaction should be accessible via the UI even with 0 confirmations (untrusted)
- [x] Adjust message content on the review tx screen
- [x] Add v2 endpoint to the SDK instead of using deprecated v1 endpoint via `getDelegates`

## How to test it

1. Open a Safe that has a delegate who is not an owner
2. Connect with that delegate
3. Observe the New transaction button is enabled
4. Observe being able to create a transaction

## Screenshots
<img width="719" alt="Screenshot 2024-07-31 at 12 39 25" src="https://github.com/user-attachments/assets/c9110ea3-089c-49c5-bf12-4dad9828016e">

## Checklist
